### DISCUSSION TITLE
~/.ssh saved as tar.bz2 attachment in keepassxc. How to restore?

### DISCUSSION THREAD

**Discussion by raghuAtRA** [twpayne/chezmoi#3110] [General]
I have my .ssh/ folder tar bzipped and saved in a keepass xc db as an attachment. I can use the `keepassxcAttachment` to get the file but how do I save it as a binary and then extract it with a `run_onchange` script? It seems whatever I do, I can't write out the file contents as a binary.

Thanks for the help!

**Comment by twpayne** [comment]
You _might_ be able to do something with a template encoding the binary contents in base64, and then somehow piping the output into a script that decodes the base64 and then piping that into `tar`, however it's likely to be complicated and messy.

I would instead store the individual files in KeePassXC, after all it's only the private key and maybe the config file that need to be kept secret.

**Reply by raghuAtRA** [reply]
tried the b64enc pipe but I couldn't get it to work; eventually, I went with your suggestion of storing individual private keys and ssh config in keepassxc.
